controller/controller.go

```diff
@@ -0,0 +1,200 @@
+package controller
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"time"
+
+	"github.com/brave/go-translate/language"
+	"github.com/brave/go-translate/translate"
+	"github.com/go-chi/chi"
+	log "github.com/sirupsen/logrus"
+)
+
+// MSTranslateServer specifies the remote MS translate server used by
+// brave-core, and it can be set to a mock server during testing.
+var MSTranslateServer = "https://api.cognitive.microsofttranslator.com"
+
+// GoogleTranslateServerProxy specifies the proxy server for requesting
+// resource from google translate server, and it can be set to a mock server
+// during testing.
+var GoogleTranslateServerProxy = "https://translate.brave.com"
+
+const (
+	// GoogleTranslateServer specifies the remote google translate server.
+	GoogleTranslateServer = "https://translate.googleapis.com"
+
+	// GStaticServerProxy specifies the proxy server for requesting resource
+	// from google gstatic server.
+	GStaticServerProxy = "https://translate-static.brave.com"
+
+	languageEndpoint = "/languages?api-version=3.0&scope=translation"
+)
+
+// TranslateRouter add routers for translate requests and translate script
+// requests.
+func TranslateRouter() chi.Router {
+	r := chi.NewRouter()
+
+	r.Post("/translate", Translate)
+	r.Get("/language", GetLanguageList)
+
+	r.Get("/translate_a/element.js", GetTranslateScript)
+	r.Get("/element/*/js/element/element_main.js", GetTranslateScript)
+	r.Get("/translate_static/js/element/main.js", GetTranslateScript)
+
+	r.Get("/translate_static/css/translateelement.css", GetGoogleTranslateResource)
+	r.Get("/images/branding/product/1x/translate_24dp.png", GetGStaticResource)
+	r.Get("/images/branding/product/2x/translate_24dp.png", GetGStaticResource)
+
+	return r
+}
+
+func getHTTPClient() *http.Client {
+	return &http.Client{
+		Timeout: time.Second * 10,
+	}
+}
+
+// GetLanguageList send a request to Microsoft server and convert the response
+// into google format and reply back to the client.
+func GetLanguageList(w http.ResponseWriter, r *http.Request) {
+	// Send a get language list request to MS
+	req, err := http.NewRequest("GET", MSTranslateServer+languageEndpoint, nil)
```

Comment:
We unset the X-Forwarded-* headers sent to Google servers at the proxy. We should unset the headers for connection to Microsoft servers as well. See: https://github.com/brave/devops/pull/785/files#diff-e262328707cf31665d8378e0949ef286R67

Reply:
they aren't automatically set. since we're not explicitly setting them on the `Request` struct I think no change is needed?

Reply:
@jumde This is a newly created request and won't have headers from google request, and I dump the request here to double check, we only have `Host: api.cognitive.microsofttranslator.com header here`.
Does this answer your concern here?